build.rs

```diff
@@ -230,13 +230,29 @@ fn parse_header_defines(header_path: impl AsRef<Path>) -> Result<HashMap<String,
     Ok(definitions)
 }
 
-fn fix_config_map(mut config_map: HashMap<String, String>) -> HashMap<String, String> {
+fn fix_config_map(
+    mut config_map: HashMap<String, String>,
+    interpreter_config: &InterpreterConfig,
+) -> HashMap<String, String> {
+    if interpreter_config.version.major == 3 && interpreter_config.version.minor.unwrap_or(0) > 7 {
+        if let Some("1") = config_map.get("Py_DEBUG").as_ref().map(|s| s.as_str()) {
+            config_map.insert("Py_REF_DEBUG".to_owned(), "1".to_owned());
+        }
+        if let Some("1") = config_map.get("Py_TRACE_REFS").as_ref().map(|s| s.as_str()) {
+            config_map.insert("Py_TRACE_REFS".to_owned(), "1".to_owned());
+        }
```

Comment:
Can you provide some details on why this change is needed?

Reply:
py >= 3.9, remove the COUNT_ALLOCS; for  COUNT_ALLOCS, there is a Cflag option -DCOUNT_ALLOCS to enable it .
py >= 3.8, enabled Py_DEBUG will   no longer enable the trace refs, meaning enabled Py_DEBUG with disabled trace refs.   If we need to trace the refs , we need configure with-trace-refs.

Reply:
Where are these changes documented?

Reply:
I've been looking at the special builds doc for each version, e.g. https://github.com/python/cpython/blob/3.8/Misc/SpecialBuilds.txt

Also in the "what's new" documents - e.g. `COUNT_ALLOCS` removal is here: https://docs.python.org/3/whatsnew/3.9.html#build-changes